src/app/forms/custom-sections/tyres/tyres.component.ts

```diff
@@ -41,24 +55,90 @@ export class TyresComponent implements OnInit, OnDestroy, OnChanges {
   get template() {
     switch (this.vehicleTechRecord.vehicleType) {
       case VehicleTypes.PSV:
-        return getTyresSection(VehicleTypes.PSV);
+        return tyresTemplatePsv;
       case VehicleTypes.HGV:
-        return getTyresSection(VehicleTypes.HGV);
+        return tyresTemplateHgv;
       case VehicleTypes.TRL:
-        return getTyresSection(VehicleTypes.TRL);
+        return tyresTemplateTrl;
     }
   }
 
+  get isPsv(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.PSV;
+  }
+
+  get isHgvOrTrl(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.HGV || this.vehicleTechRecord.vehicleType === VehicleTypes.TRL;
+  }
+
   get types(): typeof FormNodeEditTypes {
     return FormNodeEditTypes;
   }
 
+  get widths(): typeof FormNodeWidth {
+    return FormNodeWidth;
+  }
+
+  get speedCategorySymbol(): MultiOptions {
+    return speedCategorySymbolAsOptions();
+  }
+
+  get fitmentCode(): MultiOptions {
+    return fitmentCodeAsOptions();
+  }
+
   get axles(): CustomFormArray {
     return this.form.get(['axles']) as CustomFormArray;
   }
 
+  getAxleTyres(i: number): CustomFormGroup {
+    return this.axles.get([i, 'tyres']) as CustomFormGroup;
+  }
+
+  getTyresRefData(tyre: Tyres, axleNumber: number) {
+    this.isError = false;
+    this.referenceDataService
+      .fetchReferenceData(ReferenceDataResourceType.Tyres, String(tyre.tyreCode))
+      .pipe(take(1))
+      .subscribe({
+        next: data => {
+          const newTyre = {
+            tyreSize: (data as ReferenceDataTyre).tyreSize,
+            plyRating: (data as ReferenceDataTyre).plyRating,
+            dataTrAxles:
+              tyre.fitmentCode === FitmentCode.SINGLE
+                ? Number((data as ReferenceDataTyre).loadIndexSingleLoad)
+                : Number((data as ReferenceDataTyre).loadIndexTwinLoad),
+            speedCategorySymbol: tyre.speedCategorySymbol,
+            fitmentCode: tyre.fitmentCode,
+            tyreCode: tyre.tyreCode
+          };
+
+          this.vehicleTechRecord = cloneDeep(this.vehicleTechRecord);
+          this.vehicleTechRecord.axles.find(ax => ax.axleNumber === axleNumber)!.tyres = newTyre;
+          this.form.patchValue(this.vehicleTechRecord, { emitEvent: false });
+        },
+        error: _err => {
+          this.errorMessage = 'Cannot find data of this tyre';
+          this.isError = true;
+
+          const emptyTyre = {
+            tyreSize: null,
+            plyRating: null,
+            dataTrAxles: null,
+            speedCategorySymbol: tyre.speedCategorySymbol,
+            fitmentCode: tyre.fitmentCode,
+            tyreCode: tyre.tyreCode
+          };
+          this.vehicleTechRecord = cloneDeep(this.vehicleTechRecord);
```

Comment:
duplicate of 117 - 119, you could even pass in the new tyre and do the nullcheck inside